### DISCUSSION TITLE
Web3Modal v2 launch

### DISCUSSION THREAD

**Discussion by chadyj** [WalletConnect/walletconnect-monorepo#1497] [Announcements]
Say hello to Web3Modal v2.0 🔥

Refreshed with a sick design for a whole-new experience.

We took on this upgrade with the mission to create a better experience for developers and end users alike. The new Web3Modal's features include:

▪️ Refreshed, customizable UI
▪️ Support for hundreds of wallets
▪️ End user onboarding
▪️ Customizable T&Cs 
▪️ ENS domain resolution

And that's not all! We'll be adding and enhancing more in the coming weeks.

▪️ Support for multiple frameworks, starting with React and Vanilla
▪️ Pre-built components for features such as account address and real-time balance 
▪️ Multi-chain support, incl. Solana and non-EVM chains

We're also working on a dedicated Web3Modal website with interactive documentation, to make the entire experience even smoother. Until then...

Dive in. Explore. And let us know what you think! 

👉 [https://github.com/WalletConnect/web3modal](https://t.co/c2ww0n022H)

**Comment by nitindme** [comment]
Hi,

I was looking at the Web3Modal features and documentation. It is really impressive what you have. I was just wondering if you have have any support for angular framework yet or any documentation around it?

**Reply by finessevanes** [reply]
You can use this package https://docs.walletconnect.com/2.0/web3modal/html-js/installation!

**Reply by DaoudMosta** [reply]
> You can use this package https://docs.walletconnect.com/2.0/web3modal/html-js/installation!

Not work with angular Because @wagmi library

**Comment by aliburzhi** [comment]
does metamask supports Web3Modal v2 ?

**Reply by mubshrx** [reply]
Not yet

**Comment by LogeswaranA** [comment]
Does "D'cent" wallet supports web3Modal V2?, we are facing issues. It is keep loading.

Thanks

**Comment by can2112** [comment]
can we customize  the Web3NetworkSwitch button
